lib/pure/collections/sequtils.nim

```diff
@@ -750,6 +750,36 @@ macro mapLiterals*(constructor, op: untyped;
   ## works for nested tuples of arrays of sets etc.
   result = mapLitsImpl(constructor, op, nested.boolVal)
 
+iterator flatItems*[T](s: openarray[T]): auto {.noSideEffect.}=
+  ## Iterates over a deep nested arrays or sequences and returns values in order.
+  ##
+  ## Example:
+  ##
+  ## .. code-block:
+  ##   # A seq of arrays of seqs of arrays
+  ##   let numbers = @[[@[[1],[2]], @[[3], [4]]], [@[[5], [6]], @[[7], [8]]], [@[[9], [10]], @[[11], [12]]]]
+  ##   for n in flatitems(s):
+  ##     echo n
```

Comment:
Can we turn this into a ``doAssert``? :)

Reply:
I've added `doAssert` to the `flatten` proc but I think for the iterator it will render the example worse

```Nim
let numbers = @[[@[[1],[2]], @[[3], [4]]], [@[[5], [6]], @[[7], [8]]], [@[[9], [10]], @[[11], [12]]]]
let check = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
var i = 1 
for n in flatitems(s):
  doAssert n == i
  i += 1
```